docs/docs/adrs/adr-018-fault-resolutions.md

```diff
@@ -0,0 +1,101 @@
+---
+sidebar_position: 18
+title: ADR Fault Resolutions
+---
+# ADR 018: Fault Resolutions
+
+## Changelog
+* 17th July 2024: Initial draft
+
+## Status
+
+Proposed
+
+## Context
+
+Partial Set Security ([PSS](./adr-015-partial-set-security.md)) allows a subset of a provider chain's validator set to secure a consumer chain.
+ While this shared security scheme has many advantages, it comes with a risk known as the
+ [subset problem](https://informal.systems/blog/replicated-vs-mesh-security#risks-of-opt-in-security-also-known-as-ics-v-2).
+ This problem arises when a malicious majority of validators from the provider chain collude and misbehave on a consumer chain.
+ This threat is particularly relevant for Opt-in chains since they might be secured by a relatively small subset of the provider's validator set.  
+
+In cases of collusion, various types of misbehaviour can be performed by the validators, such as:
+
+* Incorrect executions to break protocol rules in order to steal funds.
+* Liveness attacks to halt the chain or censor transactions.
+* Oracle attacks to falsify information used by the chain logic.
+
+Currently, these types of attacks aren't handled in PSS, leaving the malicious validators unpunished.
+
+A potential solution is to use fraud proofs. This technology allows proving incorrect state transitions of a chain without a full node.
+ However, this is a complex technology, and there is no framework that works for Cosmos chains to this day.
+
+
+To address this risk in PSS, a governance-gated slashing solution can be used until fraud proof technology matures.
+
+
+This ADR proposes a fault resolution mechanism, which is a type of governance proposal that victims of faults can use to vote on the
+ slashing of validators that misbehave on Opt-in consumer chains.
+
+In what follows, we describe the implementation of a fault resolution mechanism that handles incorrect executions on consumer chains,
```

Comment:
Because we only provide a guideline for incorrect execution fault definition in the intersubjective faults CHIPs. The handling for other types of faults will be implemented iteratively.

Reply:
but that will not need any new implementation effort, right?

Reply:
I would recommend making this a bit more general. From a code perspective, this handles arbitrary fault resolutions

Reply:
As we all agree the ADR is specific to incorrect executions and that should be reflected in its title.
The filename will be renamed in the last step so as not to confuse the review process.